src/components/button/Button.tsx

```diff
@@ -1,66 +1,31 @@
 /** @jsxImportSource @emotion/react */
-import { css } from '@emotion/react';
-import type { CSSProperties, ReactNode } from 'react';
+import {
+  Button as BlueprintButton,
+  AnchorButton as BlueprintAnchorButton,
+  ButtonProps as BlueprintButtonProps,
+  AnchorButtonProps as BlueprintAnchorButtonProps,
+  Tooltip,
+  TooltipProps,
+} from '@blueprintjs/core';
 
-interface Colorhover {
-  basic: CSSProperties['backgroundColor'];
-  hover?: CSSProperties['backgroundColor'];
-}
-
-interface ButtonProps {
-  children: ReactNode;
-  backgroundColor?: Colorhover;
-  color?: Colorhover;
-  onClick?: () => void;
-  style?: CSSProperties;
-  disabled?: boolean;
-}
+type BlueprintProps = {
+  [key in keyof BlueprintButtonProps &
+    keyof BlueprintAnchorButtonProps]?: BlueprintButtonProps[key] &
+    BlueprintAnchorButtonProps[key];
+};
+export type ButtonProps = BlueprintProps & {
+  tooltipProps?: Omit<TooltipProps, 'children'>;
+};
 
 export function Button(props: ButtonProps) {
-  const {
-    backgroundColor = {
-      basic: 'hsl(21deg, 90%, 48%)',
-      hover: 'hsl(21deg, 90%, 40%)',
-    },
-    color = { basic: 'white', hover: 'white' },
-    style,
-    onClick,
-    children,
-    disabled = false,
-  } = props;
+  const { tooltipProps, children, ...buttonProps } = props;
 
+  const InnerButton = buttonProps.disabled
+    ? BlueprintAnchorButton
+    : BlueprintButton;
   return (
-    <button
-      style={style}
-      onClick={onClick}
-      css={css({
-        display: 'flex',
-        alignItems: 'center',
-        paddingLeft: 10,
-        paddingRight: 10,
-        paddingTop: 6,
-        paddingBottom: 6,
-        borderWidth: 1,
-        borderColor: 'transparent',
-        fontSize: '1.125em',
-        borderRadius: 6,
-        color: color.basic,
-        minWidth: 30,
-        justifyContent: 'center',
-        backgroundColor: backgroundColor.basic,
-        ':hover:enabled': {
-          color: color.hover,
-          backgroundColor: backgroundColor.hover,
-        },
-        ':disabled': {
-          opacity: 0.5,
-          cursor: 'not-allowed',
-        },
-      })}
-      type="button"
-      disabled={disabled}
-    >
-      {children}
-    </button>
+    <InnerButton {...buttonProps}>
+      <Tooltip {...tooltipProps}>{children}</Tooltip>
+    </InnerButton>
```

Comment:
Check that the tooltip is really activated on the whole button, also when bringing the cursor close to the edges.

Reply:
resolved